### PR TITLE
IGNITE-6879 and migration to spring-data 2.0.5.RELEASE

### PR DESCRIPTION
**Pull request by homich1991**
Fixed IGNITE-6879
Migrated to spring-data 2.0.5.RELEASE
Migrated to spring 5.0.4.RELEASE
Fixed overrided methods naming
----------------------------------------------------------

was problem with merging master in previous one https://github.com/apache/ignite/pull/3628, so created this one

### CODE REVIEW

File: modules/spring-data-2.0/src/main/java/org/apache/ignite/springdata/repository/query/IgniteQueryGenerator.java [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+        return new IgniteQuery(sql.toString(), parts.isCountProjection(), getOptions(mtd));
+    }
+
+    /**
+     * Add a dynamic part of query for the sorting support.
+     *
+     * @param sql SQL text string.
+     * @param sort Sort method.
+     */
+    public static StringBuilder addSorting(StringBuilder sql, Sort sort) {
```
Comment (SomeFire) [inline modules/spring-data-2.0/src/main/java/org/apache/ignite/springdata/repository/query/IgniteQueryGenerator.java]: Missing return parameter.

File: modules/spring-data-2.0/src/main/java/org/apache/ignite/springdata/repository/query/IgniteQueryGenerator.java [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+/**
+ * Ignite query generator for Spring Data framework.
+ */
+public class IgniteQueryGenerator {
+
+    /**
+     * @param mtd Method.
+     * @param metadata Metadata.
+     */
+    @NotNull public static IgniteQuery generateSql(Method mtd, RepositoryMetadata metadata) {
```
Comment (SomeFire) [inline modules/spring-data-2.0/src/main/java/org/apache/ignite/springdata/repository/query/IgniteQueryGenerator.java]: Missing return parameter.

File: modules/spring-data-2.0/src/main/java/org/apache/ignite/springdata/repository/query/IgniteRepositoryQuery.java [hunk 3]
```diff
@@ -0,0 +1,306 @@
...
+    /** {@inheritDoc} */
+    @Override public QueryMethod getQueryMethod() {
+        return new QueryMethod(mtd, metadata, factory);
+    }
+
+    /**
+     * @param mtd Method.
+     * @param isFieldQry Is field query.
+     */
+    private ReturnStrategy calcReturnType(Method mtd, boolean isFieldQry) {
```
Comment (SomeFire) [inline modules/spring-data-2.0/src/main/java/org/apache/ignite/springdata/repository/query/IgniteRepositoryQuery.java]: Missing return parameter in javadoc.

File: modules/spring-data-2.0/src/main/java/org/apache/ignite/springdata/repository/query/IgniteRepositoryQuery.java [hunk 4]
```diff
@@ -0,0 +1,306 @@
...
+            return ReturnStrategy.CACHE_ENTRY;
+        else
+            return ReturnStrategy.ONE_VALUE;
+    }
+
+    /**
+     * @param cls Class 1.
+     * @param mtd Method.
+     */
+    private boolean hasAssignableGenericReturnTypeFrom(Class<?> cls, Method mtd) {
```
Comment (SomeFire) [inline modules/spring-data-2.0/src/main/java/org/apache/ignite/springdata/repository/query/IgniteRepositoryQuery.java]: Missing return parameter in javadoc.

File: modules/spring-data-2.0/src/main/java/org/apache/ignite/springdata/repository/query/IgniteRepositoryQuery.java [hunk 5]
```diff
@@ -0,0 +1,306 @@
...
+        }
+
+        return false;
+    }
+
+    /**
+     * @param prmtrs Prmtrs.
+     * @param qryCursor Query cursor.
+     */
+    @Nullable private Object transformQueryCursor(Object[] prmtrs, QueryCursor qryCursor) {
```
Comment (SomeFire) [inline modules/spring-data-2.0/src/main/java/org/apache/ignite/springdata/repository/query/IgniteRepositoryQuery.java]: Missing return parameter in javadoc.

File: modules/spring-data-2.0/src/test/java/org/apache/ignite/springdata/IgniteSpringDataQueriesSelfTest.java [hunk 6]
```diff
@@ -0,0 +1,291 @@
...
+    /** Repository 2. */
+    private static PersonSecondRepository repo2;
+
+    /** Context. */
+    private static AnnotationConfigApplicationContext ctx;
+
+    /** Number of entries to store */
+    private static int CACHE_SIZE = 1000;
+
+    @Override protected void beforeTestsStarted() throws Exception {
```
Comment (SomeFire) [inline modules/spring-data-2.0/src/test/java/org/apache/ignite/springdata/IgniteSpringDataQueriesSelfTest.java]: Missing javadoc.

File: modules/spring-data-2.0/src/test/java/org/apache/ignite/springdata/IgniteSpringDataQueriesSelfTest.java [hunk 7]
```diff
@@ -0,0 +1,291 @@
...
+
+        repo = ctx.getBean(PersonRepository.class);
+        repo2 = ctx.getBean(PersonSecondRepository.class);
+
+        for (int i = 0; i < CACHE_SIZE; i++)
+            repo.save(i, new Person("person" + Integer.toHexString(i),
+                "lastName" + Integer.toHexString((i + 16) % 256)));
+    }
+
+    @Override protected void afterTestsStopped() throws Exception {
```
Comment (SomeFire) [inline modules/spring-data-2.0/src/test/java/org/apache/ignite/springdata/IgniteSpringDataQueriesSelfTest.java]: Missing javadoc.

File: modules/spring-data-2.0/src/test/java/org/apache/ignite/testsuites/IgniteSpringData2TestSuite.java [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+    public static TestSuite suite() throws Exception {
+        TestSuite suite = new TestSuite("Spring Data 2.0 Test Suite");
+
+        suite.addTestSuite(IgniteSpringDataCrudSelfTest.class);
+        suite.addTestSuite(IgniteSpringDataQueriesSelfTest.class);
+
+        return suite;
+    }
+}
+
```
Comment (SomeFire) [inline modules/spring-data-2.0/src/test/java/org/apache/ignite/testsuites/IgniteSpringData2TestSuite.java]: Files must end with only one empty line.

File: modules/spring-data-2.0/src/main/java/org/apache/ignite/springdata/repository/query/IgniteQueryGenerator.java [hunk 9]
```diff
@@ -0,0 +1,246 @@
...
+     * Determines whether query is dynamic or not (by list of method parameters)
+     *
+     * @param mtd Method.
+     * @return type of options
+     */
+    public static IgniteQuery.Option getOptions(Method mtd) {
+        IgniteQuery.Option option;
+
+        Class<?>[] types = mtd.getParameterTypes();
+        Class<?> type = types[types.length - 1];
```
Comment (sk0x50) [inline modules/spring-data-2.0/src/main/java/org/apache/ignite/springdata/repository/query/IgniteQueryGenerator.java]: Method#getParameterTypes() returns an an array of length 0 if the underlying executable takes no parameters.
So, the next line may result in ArrayIndexOutOfBound exception.

Please take a look at this commit https://github.com/apache/ignite/commit/6875ac3e069358af43586ad7068cfb7c50d8c08f

**Pull request closed without merging**